Common/Serialize/Serializer.cpp

```diff
@@ -111,14 +111,22 @@ void Do(PointerWrap &p, std::string &x) {
 }
 
 void Do(PointerWrap &p, std::wstring &x) {
-	int stringLen = sizeof(wchar_t)*((int)x.length() + 1);
+	int stringLen = sizeof(wchar_t) * ((int)x.length() + 1);
 	Do(p, stringLen);
 
+	auto read = [&]() {
+		std::wstring r;
```

Comment:
Would it be simpler to read into `x` directly, since it already is the right length (from what I can gather)?

So something like:

```c++
memcpy(&x[0], *p.ptr, x.size() * sizeof(wchar_t))
```

Reply:
No, because this is used for the verify path too.  We cannot overwrite x in that case.

-[Unknown]

Reply:
Could pass the string as a parameter to avoid an allocation in the read case if this is performance-sensitive but I'm guessing it's not